src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -42,7 +43,7 @@
 
   private static final String PROXY_SCHEME = "http";
   private static final int MAX_RETRIES = 3;
-
+  private static final int TIMEOUT = 300; // 5 minutes
```

Comment:
nit: CONNECTION_TIMEOUT

Reply:
Is 5 minutes too long a time? What does JDBC use?

Reply:
they use 5 mins for Socket timeout, and the other two are 1 min

Reply:
I think 5 mins is too long a time for waiting. 
Also, https://github.com/snowflakedb/snowflake-jdbc/blob/69f560b9370cbe5de112f259600911ccd74333bf/src/main/java/net/snowflake/client/jdbc/telemetryOOB/TelemetryService.java

Reply:
They also have the timeout in the util, https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/core/HttpUtil.java

Reply:
Got it! Thanks! I think it is okay for us to start with this timeout. 
CC: @sfc-gh-rramachandran